akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -518,17 +518,27 @@ private[http] object HttpServerBluePrint {
       def with100ContinueTrigger[T <: ParserOutput](createEntity: EntityCreator[T, RequestEntity]) =
         StreamedEntityCreator {
           createEntity.compose[Source[T, NotUsed]] {
-            _.via(Flow[T].transform(() ⇒ new PushPullStage[T, T] {
-              private var oneHundredContinueSent = false
-              def onPush(elem: T, ctx: Context[T]) = ctx.push(elem)
-              def onPull(ctx: Context[T]) = {
-                if (!oneHundredContinueSent) {
-                  oneHundredContinueSent = true
-                  emit100ContinueResponse.invoke(())
+            _.via(new GraphStage[FlowShape[T, T]] {
+              val in: Inlet[T] = Inlet("GraphStage.in")
+              val out: Outlet[T] = Outlet("GraphStage.out")
+              override val shape: FlowShape[T, T] = FlowShape(in, out)
+
+              override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+                new GraphStageLogic(shape) with InHandler with OutHandler {
+                  private var oneHundredContinueSent = false
+
+                  override def onPush(): Unit = push(out, grab(in))
+                  override def onPull(): Unit = {
+                    if (!oneHundredContinueSent) {
+                      oneHundredContinueSent = true
+                      emit100ContinueResponse.invoke(())
+                    }
+                    pull(in)
+                  }
+
+                  setHandlers(in, out, this)
                 }
-                ctx.pull()
-              }
-            }).named("expect100continueTrigger"))
+            }.named("expect100continueTrigger"))
```

Comment:
Since here type parameter T is used, I cannot just simply make it an object. If use the upper bound `ParserOutput` as the type variable of FlowShape(i.e. `GraphStage[FlowShape[ParserOutput, ParserOutput]]`), compiler will complain the expression of type doesn't conform to the expected type.

Any suggestion?